vent/main.py

```diff
@@ -18,16 +21,34 @@ def parse_cmd_args():
                         action='store_true')
     return parser.parse_args()
 
+def set_valves_save_position():
+    print("Terminating program; closing vents...")
+    if not args.simulation:
+        time.sleep(0.01)
+        HAL = io.Hal( config_file = 'vent/io/config/devices.ini')
+        for i in range(10):
+            HAL.setpoint_in = 0
+            HAL.setpoint_ex = 1 
+            time.sleep(0.01)
+    else:
+        print("Terminating simulation.")
 
 def main():
     args = parse_cmd_args()
-    coordinator = get_coordinator(single_process=args.single_process, sim_mode=args.simulation)
-    app, gui = launch_gui(coordinator)
-    sys.exit(app.exec_())
-
-
+    try:
```

Comment:
@sneakers-the-rat Unclear to me whether this should be here?

Reply:
I think this should be done as part of a HAL closing routine, makes sense that an assurance that hardware returns to an off state would be done in the hardware controlling object. not a strong opinion though